vscode-lean4/src/leanclient.ts

```diff
@@ -341,6 +343,9 @@ export class LeanClient implements Disposable {
     }
 
     async openLean4Document(doc: TextDocument) {
+        // Note: we must not send the DidOpenTextDocumentNotification more
+        // than once for each document otherwise we'll end up with
+        // multiple "--workers" for the same doc.
```

Comment:
For anybody else who's wondering, this is following the LSP spec:
> An open notification must not be sent more than once without a corresponding close notification send before. This means open and close notification must be balanced and the max open count for a particular textDocument is one.

Reply:
I clarified the comment (and moved it to the didOpen middleware) thanks.